### ISSUE TITLE
Improve handling of errors in contrib

### ISSUE THREAD

**Issue by Kircheneer** [nautobot/nautobot-app-ssot#628]
### Proposed Functionality

Upon failed object validation in `create`, `update`, or `delete`, raise the appropriate `diffsync.exceptions.ObjectCrudException` subclass (`ObjectNotCreated`, `ObjectNotUpdated`, `ObjectNotDeleted`). Include as much context as possible so the error can be acted upon.

### Use Case

Instead of crashing the entire SSoT run, this singles out failures in individual CRUD operations and allows them to be properly logged.

This also enables the [data validation engine integration](https://github.com/nautobot/nautobot-app-ssot/issues/96).

**Comment by Kircheneer** [comment]
On second thought, in `nautobot_ssot.contrib` this is already happening, and that is all we can enforce. We should however, document the existence of this mechanism and what it entails.

**Comment by Renrut5** [comment]
@jdrew82, suggest we close this as a non-issue for the time being unless something specific comes up. Thoughts?

**Comment by jdrew82** [comment]
I agree that at a minimum we should document the behavior.

**Comment by Renrut5** [comment]
@jdrew82 , agreed there too, I'm going to close this one out and have a more specific ticket on updating docs. There are a few other documentation items I think need to get updated and will add it to that.
